src/utils.jl

```diff
@@ -176,6 +176,34 @@ function _get_json_or(
     haskey(json, key) ? _get_json(json, key, T; msg) : default
 end
 
+# _get_json_convert is a _get_json-type helper that also does some sort of type conversion
+# parsing etc. The general signature is the following:
+#
+#   function _get_json_convert(
+#       json::Dict, key::AbstractString, ::Type{T};
+#       msg::Union{AbstractString, Nothing}=nothing
+#   )::T
+#
+# Although in practice we implement for each type separately, since the parsing/conversion logic
+# can vary dramatically.
+#
+# A key point, though, is that it will throw a JuliaHubError if the server response is somehow
+# invalid and we can't parse/convert it properly.
```

Comment:
Is there any reason to keep this separate from `_get_json`? The only difference is that `_get_json` currently does a `convert(T, value)`, whereas this uses `parse(T, value)`, which doesn't seem semantically different enough imho (especially since `convert(UUID, ::String)` isn't implemented).

Reply:
`_get_json` doesn't convert (I believe..). It does an `isa` check first, and throws if that doesn't match, so it would never get to the `convert` call (from the function return type assertion), regardless of whether the `convert` method exists or not. So in that sense, `_get_json` is restricted to only returning the basic JSON types (string, int, array etc.). 

Now whether that distinction is useful, I am not entirely sure either. But I wasn't 100% comfortable changing the semantics of `_get_json`, and so opted to make this explicit (for now; this can always be changed; also, we should really be using JSON3/StructTypes, but that's a relatively large refactor).